qutip/tests/test_random.py

```diff
@@ -91,5 +91,26 @@ def test_rand_super_dims():
         check_func_dims(func, (7, ), {}, [[[7], [7]]] * 2)
         check_func_dims(func, (6, ), {'dims': [[[2, 3], [2, 3]], [[2, 3], [2, 3]]]}, [[[2, 3], [2, 3]], [[2, 3], [2, 3]]])
 
+def test_rand_ket():
+    """
+    Random Qobjs: Tests that N and dims are determined as needed.
+    """
```

Comment:
Yeah, I saw that function but it was also testing `rand_ket_haar`, I thought it would be better to define a separate function for `rand_ket` only. Since you are requesting a change to `rand_ket_haar` as well, I will just move the extra cases there.